scripts/lib/releaseNotes.ts

```diff
@@ -150,3 +151,147 @@ function addNewReleaseNoteToc(releaseNotesNode: any, newVersion: string) {
     });
   }
 }
+
+/**
+ * Sorts the release notes dictionary by the patch version
+ */
+export function sortReleaseNotesVersions(markdownByPatchVersion: {
+  [id: string]: string;
+}) {
+  const markdownByPathEntries = Object.entries(markdownByPatchVersion);
+
+  const markdownByPatchSortedArray = markdownByPathEntries.sort(
+    ([version1], [version2]) => {
+      const versionPatch1 = version1.split("rc").slice(0, 1)[0];
+      const versionPatch2 = version2.split("rc").slice(0, 1)[0];
+      const comparison = version2.localeCompare(version1);
+
+      if (versionPatch1 == versionPatch2) {
+        // The release candidates should appear first
```

Comment:
I don't think this is fully considering that it is legal to have `0.45.1rc1`, i.e. release candidates for patch releases. `0.45.1rc1` needs to be seen as a newer release than `0.45.0`, but it is older than `0.45.1`. This would be good to unit test and use test-driven development for.

Reply:
Tested in the new test, and the case is handled. We only enter in the if condition if the two versions have the same patch, like `0.45.1` and `0.45.1rc1`. If the patch is different, the comparison we use is the one with localCompare.